tests/core/anthropic/_utils/test_setup_call.py

```diff
@@ -0,0 +1,139 @@
+"""Tests the `anthropic._utils.setup_call` module."""
+
+import inspect
+from unittest.mock import MagicMock, patch
+
+import pytest
+from anthropic import Anthropic
+
+from mirascope.core.anthropic._utils.setup_call import setup_call
+from mirascope.core.anthropic.tool import AnthropicTool
+
+
+@pytest.fixture()
+def mock_base_setup_call() -> MagicMock:
+    """Returns the mock setup_call function."""
+    mock_setup_call = MagicMock()
+    mock_setup_call.return_value = [MagicMock() for _ in range(3)] + [{}]
+    return mock_setup_call
+
+
+@patch(
+    "mirascope.core.anthropic._utils.setup_call.convert_message_params",
+    new_callable=MagicMock,
+)
+@patch("mirascope.core.anthropic._utils.setup_call._utils", new_callable=MagicMock)
+def test_setup_call(
+    mock_utils: MagicMock,
+    mock_convert_message_params: MagicMock,
+    mock_base_setup_call: MagicMock,
+) -> None:
+    """Tests the `setup_call` function."""
+    mock_utils.setup_call = mock_base_setup_call
+    fn = MagicMock()
+    create, prompt_template, messages, tool_types, call_kwargs = setup_call(
+        model="claude-3-5-sonnet-20240620",
+        client=None,
+        fn=fn,
+        fn_args={},
+        dynamic_config=None,
+        tools=None,
+        json_mode=False,
+        call_params={"max_tokens": 1000},
+        extract=False,
+    )
+    assert prompt_template == mock_base_setup_call.return_value[0]
+    assert tool_types == mock_base_setup_call.return_value[2]
+    assert (
+        "model" in call_kwargs and call_kwargs["model"] == "claude-3-5-sonnet-20240620"
+    )
+    assert "messages" in call_kwargs and call_kwargs["messages"] == messages
+    mock_base_setup_call.assert_called_once_with(
+        fn, {}, None, None, AnthropicTool, {"max_tokens": 1000}
+    )
+    mock_convert_message_params.assert_called_once_with(
+        mock_base_setup_call.return_value[1]
+    )
+    assert messages == mock_convert_message_params.return_value
+    assert inspect.signature(create) == inspect.signature(Anthropic().messages.create)
+
+    mock_base_setup_call.return_value[1] = [
+        {"role": "system", "content": [{"type": "text", "text": "test"}]}
+    ]
+    _, _, messages, _, call_kwargs = setup_call(
```

Comment:
if we're only testing call_kwargs here, why have `messages` as a named var?

Reply:
removed